autopush/db.py

```diff
@@ -151,12 +161,53 @@ def create_rotating_message_table(prefix="message", delta=0, date=None,
     # type: (str, int, Optional[datetime.date], int, int) -> Table
     """Create a new message table for webpush style message storage"""
     tablename = make_rotating_tablename(prefix, delta, date)
-    return Table.create(tablename,
-                        schema=[HashKey("uaid"),
-                                RangeKey("chidmessageid")],
-                        throughput=dict(read=read_throughput,
-                                        write=write_throughput),
-                        )
+
+    try:
+        table = DYNAMODB.Table(tablename)
+        if table.table_status == 'ACTIVE':  # pragma nocover
+            return table
+    except ClientError as ex:
+        if ex.response['Error']['Code'] != 'ResourceNotFoundException':
+            # If we hit this, our boto3 is misconfigured and we need to bail.
+            raise ex  # pragma nocover
+    table = DYNAMODB.create_table(
+        TableName=tablename,
+        KeySchema=[
+            {
+                'AttributeName': 'uaid',
+                'KeyType': 'HASH'
+            },
+            {
+                'AttributeName': 'chidmessageid',
+                'KeyType': 'RANGE'
+            }],
+        AttributeDefinitions=[{
+                'AttributeName': 'uaid',
+                'AttributeType': 'S'
+            },
+            {
+                'AttributeName': 'chidmessageid',
+                'AttributeType': 'S'
+            }],
+        ProvisionedThroughput={
+            'ReadCapacityUnits': read_throughput,
+            'WriteCapacityUnits': write_throughput
+        })
+    table.meta.client.get_waiter('table_exists').wait(
+        TableName=tablename)
+    try:
+        table.meta.client.update_time_to_live(
+            TableName=tablename,
+            TimeToLiveSpecification={
+                'Enabled': True,
+                'AttributeName': 'expry'
+            }
+        )
+    except ClientError as ex:  # pragma nocover
+        if "UnknownOperationException" in ex.message:
```

Comment:
I dunno. We're eating an error. I'm not a huge fan of this either, but the other option is that we introduce a potentially complicated test environment check system (mock, env check, passed flag even for integration, etc.). The try block is limited to the one function and the significant return, so I think it may be OK to do this.

Reply:
Yea, probably not totally necessary. I think UnknownOperationException is a pretty clear case that we should not ever see in prod. It does show up as the exception "type" (`response['Error']['Code'] == 'UnknownOperationException'`) so we can check that a little more explicitly here at least